### PR TITLE
Structure_Engine: Bug fixes for to Description and DescriptionOrName Comparer

### LINKED ISSUE #1692: Structure_Engine: round DescriptionOrName() output
#### Description:
Description outputs a formatted string based on the object's properties, which are sometimes large floating point numbers (especially for US sections and materials). These should be rounded to some sensible tolerance or be represented in scientific notation for large numbers.

### LINKED ISSUE #1701: Structure_Engine: Description crashes for null vectors
#### Description:

Vectors being null when getting description should state "null" instead of crashing.

### PR DESCRIPTION
**Pull request by peterjamesnugent**
### Issues addressed by this PR

Closes #1692 
Closes #1701 

### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/EqksKZqBSrZDu9lQuAgxGYEBGon959FhudAQ9EKEpWmMdg?e=XLa9dO

### Changelog

- Added check for null `Vector` when using `Description()`
- Added rounded numbers to 3 significant figures for `Description()` method - this will default to scientific notation if the string is shorter than the rounded equivalent in decimals

### Additional comments

You may want to test different properties for `Description()` than are available in the script.

### CODE REVIEW

File: Structure_Engine/Query/Description.cs [hunk 1]
```diff
@@ -398,7 +406,7 @@ public static string Description(this LinkConstraint constraint)
         [Output("desc", "The generated description for the constraint.")]
         public static string Description(this Offset offset)
         {
-            return offset.Start.VectorComponents() + " - " + offset.End.VectorComponents();
+            return $"{offset.Start.VectorComponents()} - + {offset.End.VectorComponents()}";
```
Comment (IsakNaslundBh) [inline Structure_Engine/Query/Description.cs:409]:
```suggestion
            return $"{offset.Start.VectorComponents()} - {offset.End.VectorComponents()}";
```

File: Structure_Engine/Query/Description.cs [hunk 2]
```diff
@@ -528,29 +536,29 @@ private static string CheckGetMaterialName(IMaterialFragment material)
...
+                $"α: { orthoTropicMaterial.ThermalExpansionCoeff.VectorComponents()}," +
+                $"ζ: { orthoTropicMaterial.DampingRatio:G3}";
         }
 
         /***************************************************/
 
         private static string VectorComponents(this Vector vector)
         {
-            return "[" + vector.X + "," + vector.Y + "," + vector.Z + "]";
+            return (vector != null) ? $"[{vector.X:G3}, {vector.Y:G3}, {vector.Z:G3}]" : null;
```
Comment (IsakNaslundBh) [inline Structure_Engine/Query/Description.cs:561]:
```suggestion
            return (vector != null) ? $"[{vector.X:G3}, {vector.Y:G3}, {vector.Z:G3}]" : "null";
```

**Comment by FraserGreenroyd** [comment]
This PR needs a title for which engine for the change log please

**Comment by IsakNaslundBh** [comment]
Good catch @FraserGreenroyd . Updated.

**Pull request merged**